OsmAnd-java/src/main/java/net/osmand/search/core/SearchResult.java

```diff
@@ -127,6 +134,17 @@ private int getSelfWordCount() {
 		return inc;
 	}
 
+	private boolean checkPhraseIsObjectType(String searchPhraseName) {
+		if (object instanceof Amenity) {
```

Comment:
not correct implementation cause it doesn't cover all cases of search should be refatored